### PR TITLE
Awesome Books: Milestone 1

### PR DESCRIPTION
**Pull request by sireclifford**
- Create a collection that keeps a list of books.
- Create a function to add a new book to the collection, with title and author.
- Create a function to remove a book from the collection
- Display all books saved in the collection in the top part of the page.
- Make sure that when a user clicks the "Add" button: a new book is added to the collection & the new book is displayed in the page.
- Make sure that when a user clicks the "Remove" button: the correct book is removed from the collection & the correct book disappears from the page.
- Make sure that data is preserved in the browser's memory by using [localStorage]

### CODE REVIEW

File: assets/scripts/index.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    bookList.innerHTML = books;
+  } else {
+    console.log('No books found');
+  }
+};
+
+window.onload = () => {
+  retrieveBooks();
+};
+const booksArray = [];
```
Comment (ShahierNashaat) [inline assets/scripts/index.js]: - [x] Kindly be noticed that you should make get the data from the local storage and put it in the booksArray so when you add data to booksArray it should append to the data in the local storage ie: 
`const booksArray = JSON.parse(localStorage.getItem('books'))? JSON.parse(localStorage.getItem('books')) : [];`

File: assets/scripts/index.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+const removeBook = (bookId) => {
+  const foundBook = JSON.parse(localStorage.getItem('books')).find(
+    (book) => book.id === bookId * 1,
+  );
+  const temp = JSON.parse(localStorage.getItem('books'));
+  const index = temp.map((object) => object.id).indexOf(foundBook.id);
+  temp.splice(index, 1);
+  localStorage.setItem('books', JSON.stringify(temp));
+  retrieveBooks();
+};
```
Comment (ShahierNashaat) [inline assets/scripts/index.js]: - [x] It is recommended to use the filter in this function and also notice that you should remove the book from booksArray or make the booksArray equal to temp.
```
const removeBook = (bookId) => {
  const temp = JSON.parse(localStorage.getItem('books')).filter(
    (book) => book.id !== bookId,
  );
  booksArray = temp; 
  localStorage.setItem('books', JSON.stringify(temp));
  retrieveBooks();
};
```

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @sireclifford , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @sireclifford  
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**